python/lsst/daf/butler/core/fileTemplates.py

```diff
@@ -79,23 +86,28 @@ def getTemplate(self, datasetTypeName):
         KeyError
             No template could be located for this Dataset type.
         """
+
+        # Get the names to use for lookup
+        names = entity.lookupNames()
```

Comment:
I'm still on the fence about this. On the one hand special casing this for `DatasetRef` would be sad, but on the other hand I don't like having `lookupNames` on `datasetRef` if it isn't actually `DatasetRef` specific. Maybe I'm just not liking the name `lookupNames`? Perhaps `configTypeNames`/`typeNamesInConfig`/`typeNames` or something would be better?

Reply:
Correct. It's allowing us to take an object with a compatible interface and let us work out which entry in the config object is relevant. I have no problem with changing the name of this method.

Reply:
Maybe make it private, since it is an implementation detail of the Butler framework and the classes that use it are effectively "friends"?